src/util/sss_krb5.c

```diff
@@ -395,16 +395,16 @@ krb5_error_code find_principal_in_keytab(krb5_context ctx,
     }
 
     if (!principal_found) {
-        kerr = KRB5_KT_NOTFOUND;
-        DEBUG(SSSDBG_TRACE_FUNC,
-              "No principal matching %s@%s found in keytab.\n",
-               pattern_primary, pattern_realm);
-        goto done;
-    }
-
-    /* check if we got any errors from krb5_kt_next_entry */
-    if (kt_err != 0 && kt_err != KRB5_KT_END) {
-        DEBUG(SSSDBG_CRIT_FAILURE, "Error while reading keytab.\n");
+        /* (!principal_found) => (kt_err != 0) */
```

Comment:
Can you make the comment a bit more explicit? At the first glance it looked a bit like commented out code.

Reply:
Fixed.